src/tests/test_fd_algorithm.cpp

```diff
@@ -124,9 +124,21 @@ TYPED_TEST_P(AlgorithmTest, ConsistentRepeatedExecution) {
     }
 }
 
+TYPED_TEST_P(AlgorithmTest, MaxLHSOptionWork) {
+    using namespace config::names;
+    unsigned int max_lhs_ = 1;
+    auto algorithm = TestFixture::CreateMaxLHSAlgorithmInstance(kTestFD, max_lhs_);
+    algorithm->Execute();
+    std::list<FD> result_fds_list = algorithm->FdList();
+    for (auto& fd : result_fds_list) {
+        ASSERT_TRUE(fd.GetLhs().GetArity() <= max_lhs_);
+    }
```

Comment:
This code verifies that all found fds has lhs size less than `max_lhs`, but this doesn't guarantee at all that fds are correct. Not sure, what is the best way to achieve this, but one approach is to verify all algorithm results via some fixed algorithm, for example, Pyro.
Also, I'd add more datasets, one small synthetic dataset doesn't look convincing for correctness verification.

Reply:
This is not yet fixed and is rather important